### PR TITLE
Data table inline editing

### PR DESCRIPTION
**Pull request by ShawnMa16**

### CODE REVIEW

File: Sources/FioriSwiftUICore/DataTable/ItemView.swift [hunk 1]
```diff
@@ -97,4 +138,109 @@ struct ItemView: View {
...
+                    .onAppear {
+                        setKeyboardNotifications()
+                    }
+                } else {
+                    Text(text ?? "")
+                }
+            }
+    }
+    
+    func setKeyboardNotifications() {
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/DataTable/ItemView.swift:189]: The notification added by `addObserver` needs to be removed at some point when the view is destroyed. 

In SwiftUI there might be a better alternative solution to add notification. See the post: https://stackoverflow.com/questions/61822307/how-to-remove-observer-in-swiftui

File: Sources/FioriSwiftUICore/DataTable/ItemView.swift [hunk 2]
```diff
@@ -86,7 +121,13 @@ struct ItemView: View {
             }
             .background(self.backgroundColor.frame(width: (((self.dataItem.isLast && self.isHeader) ? self.layoutManager.tableTrailingLayoutMargin : 0) + self.dataItem.size.width + contentInset * 2) * self.layoutManager.scaleX, height: self.dataItem.rowHeight * self.layoutManager.scaleY))
             .offset(x: (self.dataItem.isLast && self.isHeader) ? self.layoutManager.tableTrailingLayoutMargin / 2 : 0, y: 0)
+            .overlay(
+                RoundedRectangle(cornerRadius: 0)
+                    .stroke(Color.blue.opacity(self.isDetectingLongPress ? 1 : (self.inlineEditing ? 1 : 0)), lineWidth: 2)
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/DataTable/ItemView.swift:126]: All the colors used in component need to be Fiori Color `Color.preferredColor()`

File: Sources/FioriSwiftUICore/DataTable/MultiPicker.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                            self.layoutManager.model.currentPicker?.selections = self.selections
+                            if let didChangeHandler = self.layoutManager.model.currentPicker?.valueDidChange {
+                                didChangeHandler(self.selections)
+                            }
+                        }
+                        resumeTableOffset()
+                    }
+                }) {
+                    HStack {
+                        Text("Done")
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/DataTable/MultiPicker.swift:49]: Hard coded text needs to be localized.

File: Sources/FioriSwiftUICore/DataTable/TableModel.swift [hunk 4]
```diff
@@ -57,10 +57,21 @@ public class TableModel: ObservableObject {
     /// Selected Indexes.
     @Published public var selectedIndexes: [Int] = []
     
+    @Published public var inlineEditingEable: Bool = true
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/DataTable/TableModel.swift:60]: typo.

I will remove this property in table model and add a `isEditable` property in `DataItem`.

**Review by billzhou0223 — CHANGES_REQUESTED** [review]
One question need to be clarified by designer: How the picker should be presented in regular mode? Still a bottom sheet or a model sheet in the center of view?

File: Sources/FioriSwiftUICore/DataTable/DataTextItem.swift [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import SwiftUI
 
 /// Text item for `DataTable`
-public struct DataTextItem: DataItem, CheckBinding {
+public class DataTextItem: DataItem, CheckBinding {
```
[the remark this answers is not in the document]
Comment (ShawnMa16) [reply Sources/FioriSwiftUICore/DataTable/DataTextItem.swift:4]: I made this change due to the in place editing feature

**Comment by MarcoEidinger** [comment]
@billzhou0223  can this PR be closed (without merge) ?

**Comment by billzhou0223** [comment]
Not needed anymore.

**Pull request closed without merging**